packages/devtools_app/lib/src/framework/notifications.dart

```diff
@@ -110,41 +138,30 @@ class NotificationsState extends State<_NotificationsProvider>
   // TODO(peterdjlee): Support clickable links in notification text. See #2268.
   /// Pushes a notification [message], and returns whether the notification was
   /// successfully pushed.
-  @override
-  bool push(
-    String message, {
-    List<Widget> actions = const [],
-    Duration duration = Notifications.defaultDuration,
-    bool allowDuplicates = true,
-  }) {
-    if (!allowDuplicates &&
-        _notifications.isNotEmpty &&
-        _notifications.where((n) => n.message == message).isNotEmpty) {
-      return false;
-    }
+  void _push(NotificationMessage message) {
     setState(() {
       _notifications.add(
         _Notification(
           message: message,
-          actions: actions,
           remove: _removeNotification,
-          duration: duration,
         ),
       );
       _overlayEntry?.markNeedsBuild();
     });
-    return true;
   }
 
   /// Dismisses all notifications with a matching message.
-  void dismiss(String message) {
+  void _dismiss(String message) {
+    print('dismiss invoked');
     bool didDismiss = false;
     // Make a copy so we do not remove a notification from [_notifications]
     // while iterating over it.
-    final notifications = List.from(_notifications);
+    final notifications = List.from(_notifications).cast<_Notification>();
```

Comment:
can you add the type to List.from to get rid of the cast: `List<_Notifications>.from(_notifications)`